assets/src/helpers.js

```diff
@@ -31,3 +47,185 @@ export const maybeEnqueueFontStyle = ( name ) => {
+ */
+export const addAMPAttributes = ( settings, name ) => {
+	if ( ! ALLOWED_CHILD_BLOCKS.includes( name ) ) {
+		return settings;
+	}
+
+	const addedAttributes = {
+		anchor: {
+			type: 'string',
+			source: 'attribute',
+			attribute: 'id',
+			selector: '*',
+		},
+	};
+
+	// Define selector according to mappings.
+	if ( BLOCK_TAG_MAPPING[ name ] ) {
+		addedAttributes.ampAnimationType = {
+			source: 'attribute',
+			selector: BLOCK_TAG_MAPPING[ name ],
+			attribute: 'animate-in',
+		};
+		addedAttributes.ampAnimationDelay = {
+			source: 'attribute',
+			selector: BLOCK_TAG_MAPPING[ name ],
+			attribute: 'animate-in-delay',
+			default: '0ms',
+		};
+		addedAttributes.ampAnimationDuration = {
+			source: 'attribute',
+			selector: BLOCK_TAG_MAPPING[ name ],
+			attribute: 'animate-in-duration',
+		};
+		addedAttributes.ampAnimationAfter = {
+			source: 'attribute',
+			selector: BLOCK_TAG_MAPPING[ name ],
+			attribute: 'animate-in-after',
+		};
+	} else if ( 'core/list' === name ) {
+		addedAttributes.ampAnimationType = {
+			type: 'string',
+		};
+		addedAttributes.ampAnimationDelay = {
+			type: 'number',
+			default: 0,
+		};
+		addedAttributes.ampAnimationDuration = {
+			type: 'number',
+			default: 0,
+		};
+		addedAttributes.ampAnimationAfter = {
+			type: 'string',
+		};
+	}
+
+	if ( 'core/image' === name ) {
+		addedAttributes.ampShowImageCaption = {
+			type: 'boolean',
+			default: false,
+		};
+	}
+
+	return {
+		...settings,
+		attributes: {
+			...settings.attributes,
+			...addedAttributes,
+		},
+		supports: {
+			...settings.supports,
+			anchor: false,
+		},
+	};
+};
+
+/**
+ * Add extra attributes to save to DB.
+ *
+ * @param {Object} props Properties.
+ * @param {Object} blockType Block type.
+ * @param {Object} attributes Attributes.
+ * @return {Object} Props.
+ */
+export const addAMPExtraProps = ( props, blockType, attributes ) => {
+	const ampAttributes = {};
+
+	if ( ! ALLOWED_CHILD_BLOCKS.includes( blockType.name ) ) {
+		return props;
+	}
+
+	// Always add anchor ID regardless of block support. Needed for animations.
+	props.id = attributes.anchor || uuid();
+
+	if ( attributes.ampAnimationType ) {
+		ampAttributes[ 'animate-in' ] = attributes.ampAnimationType;
+
+		if ( attributes.ampAnimationDelay ) {
+			ampAttributes[ 'animate-in-delay' ] = attributes.ampAnimationDelay;
+		}
+
+		if ( attributes.ampAnimationDuration ) {
+			ampAttributes[ 'animate-in-duration' ] = attributes.ampAnimationDuration;
+		}
+
+		if ( attributes.ampAnimationAfter ) {
+			ampAttributes[ 'animate-in-after' ] = attributes.ampAnimationAfter;
+		}
+	}
+
+	if ( attributes.ampFontFamily ) {
+		ampAttributes[ 'data-font-family' ] = attributes.ampFontFamily;
+	}
+
+	return {
+		...props,
+		...ampAttributes,
+	};
+};
+
+/**
+ * Add page number label to page blocks
+ *
+ * @param {Object} BlockListBlock BlockListBlock element.
+ * @return {Function} Handler.
+ */
+export const disableBlockDropZone = createHigherOrderComponent(
```

Comment:
DropZone is still needed for other blocks.

Reply:
@miina The function only disables the drop zone when reordering, as can be seen further down. Will improve docs.